### PR TITLE
Update Rekor verification example and endorsement verification logic

### PR DESCRIPTION
**Pull request by rbehjati**
This PR:

- Moves the verify module from `oak_remote_attestation` to `oak_remote_attestation_verification`
- Updates testdata in `oak_remote_attestation_verification` with an updated endorsement statement based on the new format, and a signature generated using Google Cloud KMS.
- Adds basic logic for verifying BinaryAttestation.

@conradgrobler  @ipetr0v It is not clear to me where the public keys come from, so in `verify_binary_attestation` they are currently passed in as separate arguments, but I suppose they should be available in AttestationBundle somewhere.

For the Rekor public key, we might want to read from a file client-side, but I don't think that will work with no_std. Perhaps we could embed the bytes. But that feels very restrictive, given that the Rekor public key might change, although I guess not very frequently. 

I am also not sure about the semantics of ReferenceValue. I think we need something more expressive, and eventually something similar to https://github.com/project-oak/transparent-release/blob/main/proto/provenance_verification.proto. Mainly because, we should be able to disable some of these checks (particularly Rekor checks) for test.

### CODE REVIEW

File: oak_remote_attestation_verification/src/rekor.rs [hunk 1]
```diff
@@ -170,21 +171,20 @@ impl TryFrom<&LogEntry> for RekorSignatureBundle {
...
+/// 1. verifying the signature in `body.RekordObj.signature`, using the endorser's public key,
 /// 1. verifying that the content of the body matches the input `endorsement_bytes`.
 ///
 /// Returns `Ok(())` if the verification succeeds, otherwise returns `Err()`.
 pub fn verify_rekor_log_entry(
     log_entry_bytes: &[u8],
-    pem_encoded_public_key_bytes: &[u8],
-    oak_public_key_bytes: &[u8],
+    pem_encoded_rekor_public_key_bytes: &[u8],
+    pem_encoded_endorser_key_bytes: &[u8],
```
Comment (conradgrobler) [inline oak_remote_attestation_verification/src/rekor.rs:181]: nit: perhaps also include "public" in the name here for consistency.

**Review by conradgrobler — APPROVED** [review]

**Review by rbehjati — COMMENTED** [review]
Thanks for the review.

**Comment by ipetr0v** [comment]
> It is not clear to me where the public keys come from, so in verify_binary_attestation they are currently passed in as separate arguments, but I suppose they should be available in AttestationBundle somewhere.

Attestation evidence will contain both encryption and signing public keys.

> For the Rekor public key, we might want to read from a file client-side, but I don't think that will work with no_std. Perhaps we could embed the bytes. But that feels very restrictive, given that the Rekor public key might change, although I guess not very frequently.

We can make the `no_std` library accept key bytes, and just have an additional logic on the client device (i.e. in the `oak_client` crate or Java class) to load it from file.

> I am also not sure about the semantics of ReferenceValue. I think we need something more expressive, and eventually something similar to https://github.com/project-oak/transparent-release/blob/main/proto/provenance_verification.proto. Mainly because, we should be able to disable some of these checks (particularly Rekor checks) for test.

The reference value is not finalized yet.

File: oak_remote_attestation_verification/src/tests.rs [hunk 2]
```diff
@@ -17,40 +17,80 @@
...
+    // https://rekor.sigstore.dev/api/v1/log/publicKey.
     let rekor_pubkey_path = "testdata/rekor_public_key.pem";
 
     let endorsement_bytes = fs::read(endorsement_path).expect("couldn't read endorsement file");
     let log_entry_bytes = fs::read(log_entry_path).expect("couldn't read log entry file");
-    let rekor_pem_bytes =
+    let rekor_key_pem_bytes =
         fs::read(rekor_pubkey_path).expect("couldn't read Rekor's public key file");
-    let pubkey_pem_bytes =
+    let endorser_key_pem_bytes =
```
Comment (ipetr0v) [inline oak_remote_attestation_verification/src/tests.rs:74]: I think these variables can also have `_public_` in the name.

**Review by ipetr0v — APPROVED** [review]

File: oak_remote_attestation_verification/src/tests.rs (hunk 2 above)
Comment (rbehjati) [reply oak_remote_attestation_verification/src/tests.rs]: Thanks. Done.

**Pull request merged**